compiler/qsc_doc_gen/src/generate_docs.rs

```diff
@@ -316,102 +449,213 @@ Fully qualified name: {fqn}
-            MetadataKind::Udt => "udt",
-            MetadataKind::Export => "export",
-        };
-        write!(
-            f,
-            "---
-uid: {}
-title: {}
-ms.date: {{TIMESTAMP}}
-ms.topic: {}
-qsharp.kind: {}
-qsharp.package: {}
-qsharp.namespace: {}
-qsharp.name: {}
-qsharp.summary: \"{}\"
----",
-            self.uid,
-            self.title,
-            self.topic,
-            kind,
-            self.package,
-            self.namespace,
-            self.name,
-            self.summary
-        )
-    }
+/// Generates the Table of Contents file, toc.yml
+fn generate_toc(map: &mut ToC, files: &mut Files) {
+    let header = "
+# This file is automatically generated.
+# Please do not modify this file manually, or your changes will be lost when
+# documentation is rebuilt.";
+
+    // This inserts the top index into the table of contents.
+    // The file itself is inserted in the npm/generate_docs.js file.
+    let top_index = "- items:
+  name: Overview
+  uid: Microsoft.Quantum.apiref-toc";
+
+    let mut table = map
+        .iter_mut()
+        .map(|(namespace, items)| {
+            let items_str = items
+                .iter()
+                .map(|item| format!("  - {{name: {}, uid: {}}}", item.name, item.uid))
+                .collect::<Vec<String>>()
+                .join("\n");
+            let content =
+                format!("- items:\n{items_str}\n  name: {namespace}\n  uid: Qdk.{namespace}");
+            (namespace, content)
+        })
+        .collect::<Vec<_>>();
+
+    table.sort_unstable_by_key(|(n, _)| {
+        // Ensures that the Microsoft.Quantum.Unstable namespaces are listed last.
+        if n.starts_with("Microsoft.Quantum.Unstable") {
+            format!("1{n}")
+        } else {
+            format!("0{n}")
+        }
+    });
+    let table = table
+        .into_iter()
+        .map(|(_, c)| c)
+        .collect::<Vec<_>>()
+        .join("\n");
+    let content = format!("{header}\n{top_index}\n{table}");
+
+    let file_name: Arc<str> = Arc::from("toc.yml");
+    let file_metadata: Arc<str> = Arc::from("");
+    let file_content: Arc<str> = Arc::from(content.as_str());
+    files.push((file_name, file_metadata, file_content));
 }
 
-enum MetadataKind {
-    Function,
-    Operation,
-    Udt,
-    Export,
+fn get_namespace(package: &Package, item: &Item) -> Option<Rc<str>> {
+    match item.parent {
+        Some(local_id) => {
+            let parent = package
+                .items
+                .get(local_id)
+                .expect("Could not resolve parent item id");
+            match &parent.kind {
+                ItemKind::Namespace(name, _) => {
+                    if name.starts_with("QIR") {
+                        None // We ignore "QIR" namespaces
+                    } else {
+                        Some(name.name())
+                    }
+                }
+                _ => None,
+            }
+        }
+        None => None,
+    }
```

Comment:
I wrote this in the github comment box so it may not be correct, but I think this code can be simplified a bit:

```suggestion
    let local_id = item.parent?;
    let parent = package
        .items
        .get(local_id)
        .expect("Could not resolve parent item id");
    let ItemKind::Namespace(name, _) = &parent.kind else { return None };
    if name.starts_with("QIR") {
        None // We ignore "QIR" namespaces
    } else {
        Some(name.name())
    }

```